plugins/inputs/s7comm/type_conversions.go

```diff
@@ -13,7 +13,7 @@ func determineConversion(dtype string, extra int) converterFunc {
 	switch dtype {
 	case "X":
 		return func(buf []byte) interface{} {
-			return (buf[0] & (1 << extra)) != 0
+			return buf[0] != 0
```

Comment:
Should this be in this PR?

Reply:
Nope, your right. It is related to issue 14043. Gone remove this. This will resolve the unused param as well.